### ISSUE TITLE
Fails to build with gcc10 (narrowing conversion)

### ISSUE THREAD

**Issue by heirecka** [OpenTTD/grfcodec#5]
Trying to build 6.0.6 fails with the following error:
```
x86_64-pc-linux-gnu-c++ -c -o objs/command.o -idirafter/usr/include -O2 -DGCC32 -D_FORTIFY_SOURCE=2 -Wall -Wextra -Wno-format-nonliteral -DWITH_FMEMOPEN -DWITH_PNG -I/usr/x86_64-pc-linux-gnu/include/libpng16 -march=native -O2 -pipe -MMD -MF objs/command.o.d -MT objs/command.o src/command.cpp
src/command.cpp: In function 'bool parse_comment(const string&)':
src/command.cpp:306:9: error: narrowing conversion of '-1' from 'int' to 'unsigned int' [-Wnarrowing]
  306 |   case -1:
```

Fedora patched it:
https://src.fedoraproject.org/rpms/grfcodec/blob/master/f/gcc10.patch

**Comment by matthijskooijman** [comment]
Same problem is signaled in Debian. Fedora patch looks fine, so I'll probably include that in the Debian package too if no new grfcodec version is released with the fix in the next 2 weeks or so.

**Comment by matthijskooijman** [comment]
I opened a PR to fix this, see #6. I took a slightly different approach than Fedora did, which I think is a cleaner fix.

**Comment by matthijskooijman** [comment]
I've also backported this fix to the Debian package, so no more hurry from that side :-)
